tap_trustpilot/__init__.py

```diff
@@ -24,23 +24,27 @@ def output_schema(stream):
 
 
 def sync(ctx):
-    streams_.business_units.fetch_into_cache(ctx)
-
-    currently_syncing = ctx.state.get("currently_syncing")
-    start_idx = streams_.all_stream_ids.index(currently_syncing) \
-        if currently_syncing else 0
-    stream_ids_to_sync = [cs.tap_stream_id for cs in ctx.catalog.streams
-                          if cs.is_selected()]
-    streams = [s for s in streams_.all_streams[start_idx:]
-               if s.tap_stream_id in stream_ids_to_sync]
-
-    for stream in streams:
-        ctx.state["currently_syncing"] = stream.tap_stream_id
-        output_schema(stream)
+    bu_ids = ctx.config.get('business_unit_ids').replace(" ", "").split(",")
+    for bu_id in bu_ids:
+        LOGGER.info(f"Extracting data for business_unit_id {bu_id}")
+        ctx.config['business_unit_id'] = bu_id
+        streams_.business_units.fetch_into_cache(ctx)
+
+        currently_syncing = ctx.state.get("currently_syncing")
+        start_idx = streams_.all_stream_ids.index(currently_syncing) \
+            if currently_syncing else 0
+        stream_ids_to_sync = [cs.tap_stream_id for cs in ctx.catalog.streams
+                              if cs.is_selected()]
+        streams = [s for s in streams_.all_streams[start_idx:]
+                   if s.tap_stream_id in stream_ids_to_sync]
+
+        for stream in streams:
+            ctx.state["currently_syncing"] = stream.tap_stream_id
+            output_schema(stream)
+            ctx.write_state()
+            stream.sync(ctx)
+        ctx.state["currently_syncing"] = None
```

Comment:
If business_unit_id is invalid or doesn't have access we are raising run time error in streams.py file.

Reply:
@KrisPersonal currently, API is throwing Bad request 400 error for a invalid business_unit_id. So the extractions for other valid business_unit_ids will fail. Instead of should we log an error saying invalid business_unit_id and continue for other valid bu_ids?